.github/workflows/dev_ci_cd.yml

```diff
@@ -0,0 +1,330 @@
+          LICENSE_SERVER: ${{ secrets.LICENSE_SERVER }}
+          MECHANICAL_IMAGE: ${{ env.DOCKER_PACKAGE }}:${{ matrix.mechanical-version }}
+        run: |
+          docker pull ${{ env.MECHANICAL_IMAGE }}
+          docker run --restart always --name ${{ env.DOCKER_MECH_CONTAINER_NAME }} -e ANSYSLMD_LICENSE_FILE=1055@${{ env.LICENSE_SERVER }} -p ${{ env.PYMECHANICAL_PORT }}:10000 ${{ env.MECHANICAL_IMAGE }} > log.txt &
+          grep -q 'WB Initialize Done' <(timeout 60 tail -f log.txt)
+
+      - name: Testing
+        uses: ansys/actions/tests-pytest@v4
+        with:
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+
+      # - name: Upload coverage to Codecov
+      #   uses: codecov/codecov-action@v3
+
+      - name: Upload coverage results
+        uses: actions/upload-artifact@v3
+        if: matrix.mechanical-version == env.DOCKER_IMAGE_VERSION
+        with:
+          name: coverage-tests
+          path: .cov
+          retention-days: 7
+
+      - name: Upload coverage results (as .coverage)
+        uses: actions/upload-artifact@v3
+        if: matrix.mechanical-version == env.DOCKER_IMAGE_VERSION
+        with:
+          name: coverage-file-tests
+          path: .coverage
+          retention-days: 7
+
+      - name: Get Mechanical container logs
+        if: always()
+        run: |
+          docker logs ${{ env.DOCKER_MECH_CONTAINER_NAME }} > mechanical_tests_log-${{ matrix.mechanical-version }}.txt 2>&1
+          echo CONTAINER LOGS OUTPUT
+          cat mechanical_tests_log-${{ matrix.mechanical-version }}.txt
+          echo CPU info
+          lscpu
+
+      - name: Upload container logs
+        uses: actions/upload-artifact@v3
+        with:
+          name: mechanical_tests_log-${{ matrix.mechanical-version }}
+          path: mechanical_tests_log-${{ matrix.mechanical-version }}.txt
+          retention-days: 7
+
+  embedding-tests:
+    name: Embedding testing and coverage
+    runs-on: ubuntu-latest
+    # needs: [smoke-tests]
+    container:
+      image: ghcr.io/ansys/mechanical:${{ github.event.inputs.version }}
+      options: --entrypoint /bin/bash
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.8', '3.9', '3.10', '3.11']
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: |
+          apt update
+          apt install -y lsb-release
+          apt install -y python3-pip
+          apt install -y python3.8-venv
+          pip3 install --trusted-host pypi.org --trusted-host pypi.python.org --trusted-host files.pythonhosted.org pip setuptools
+          pip3 install --upgrade pip flit
+      - name: Install packages for testing
+        run: |
+          pip install .[tests]
+
+      - name: Unit Testing and coverage
+        env:
+          LICENSE_SERVER: ${{ secrets.LICENSE_SERVER }}
+          ANSYSCL${{ github.event.inputs.revn }}_DIR: /install/ansys_inc/v${{ github.event.inputs.revn }}/licensingclient
```

Comment:
I feel like this wouldn't work, but it won't let me test the workflow in this branch.

Reply:
That's not gonna work indeed... You should output the variable to GITHUB_ENV if you want to create "custom" named variable. See https://github.com/ansys/actions/blob/main/build-wheelhouse/action.yml#L87 for example